crates/rustc_codegen_spirv/src/linker/structurizer.rs

```diff
@@ -106,69 +105,51 @@ impl ControlFlowInfo {
-        let mut retval = Vec::new();
-
+    fn set_names(&self, builder: &mut Builder) {
         for loop_info in &self.loops {
-            retval.push((loop_info.header_id, "loop_header".to_string()));
-            retval.push((loop_info.merge_id, "loop_merge".to_string()));
-            retval.push((loop_info.continue_id, "loop_continue".to_string()));
+            builder.name(loop_info.header_id, "loop_header".to_string());
+            builder.name(loop_info.merge_id, "loop_merge".to_string());
+            builder.name(loop_info.continue_id, "loop_continue".to_string());
         }
         for id in &self.if_merge_ids {
-            retval.push((*id, "if_merge".to_string()));
+            builder.name(*id, "if_merge".to_string());
         }
         for id in &self.switch_merge_ids {
-            retval.push((*id, "switch_merge".to_string()));
+            builder.name(*id, "switch_merge".to_string());
         }
-
-        retval
     }
 }
 
-pub fn structurize(sess: &Session, module: &mut Module) {
-    let mut debug_names = Vec::new();
+pub fn structurize(sess: &Session, module: Module) -> Module {
+    let mut builder = Builder::new_from_module(module);
 
-    for func in &mut module.functions {
+    for func_idx in 0..builder.module_ref().functions.len() {
         let mut cf_info = ControlFlowInfo::new();
 
-        insert_loop_merge_on_conditional_branch(
-            &mut module.header.as_mut().unwrap(),
-            &mut func.blocks,
-            &mut cf_info,
-        );
-
-        retarget_loop_children_if_needed(&mut func.blocks, &cf_info);
-
-        insert_selection_merge_on_conditional_branch(
-            sess,
-            &mut module.header.as_mut().unwrap(),
-            &mut func.blocks,
-            &mut cf_info,
-        );
-
-        defer_loop_internals(
-            &mut module.header.as_mut().unwrap(),
-            &mut func.blocks,
-            &cf_info,
-            &mut debug_names,
-            &mut module.types_global_values,
-        );
-
-        debug_names.extend(cf_info.get_debug_names());
-    }
+        builder.select_function(Some(func_idx)).unwrap();
 
-    for (id, name) in debug_names {
-        module.debugs.push(Instruction::new(
-            Op::Name,
-            None,
-            None,
-            vec![Operand::IdRef(id), Operand::LiteralString(name)],
-        ))
+        insert_loop_merge_on_conditional_branch(&mut builder, &mut cf_info);
+        retarget_loop_children_if_needed(&mut builder, &cf_info);
+        insert_selection_merge_on_conditional_branch(sess, &mut builder, &mut cf_info);
+        defer_loop_internals(&mut builder, &cf_info);
+        cf_info.set_names(&mut builder);
     }
+
+    builder.module()
+}
+
+fn get_blocks_mut(builder: &mut Builder) -> &mut Vec<Block> {
+    let function = builder.selected_function().unwrap();
+    &mut builder.module_mut().functions[function].blocks
+}
+
+fn get_blocks_ref(builder: &Builder) -> &Vec<Block> {
```

Comment:
I've had clippy yell at me so much for this, it's just seared into my head: prefer `&[Block]` over `&Vec<Block>` (I'm guessing clippy didn't fire on this since it's a return type)

Reply:
Replaced with slice :)
`Ashley 1 - Clippy 0`.